openverse_catalog/dags/providers/provider_api_scripts/nypl.py

```diff
@@ -155,31 +160,54 @@ def _get_creators(creatorinfo):
         creator = None
 
     if creator is None:
-        logger.warning("No primary creator found")
+        logger.debug("No primary creator found")
 
     return creator
 
 
-def _get_image_url(images, image_url_dimensions=None):
-    if image_url_dimensions is None:
-        image_url_dimensions = IMAGE_URL_DIMENSIONS
-    image_type = {
-        parse_qs(urlparse(img.get("$")).query)["t"][0]: img.get("$") for img in images
+def _get_filetype(description: str):
+    """
+    Extracts the filetype from a description string like:
+    "Cropped .jpeg (1600 pixels on the long side)"
+    :param description: the description string
+    :return:  jpeg | gif
+    """
+    if match := re.search(FILETYPE_PATTERN, description):
+        return match.group(1)
+    return None
+
+
+def _get_image_data(images):
+    """
+    Gets a list of dictionaries of the following shape:
+    {
+      "$": "http://images.nypl.org/index.php?id=56738467&t=q&download=1
+    &suffix=29eed1f0-3d50-0134-c4c7-00505686a51c.001",
+      "description": "Cropped .jpeg (1600 pixels on the long side)"
     }
+    Extracts the largest image based on the `t` query parameter
+    and IMAGE_URL_DIMENSIONS.
+    """
 
-    image_url = _get_preferred_image(image_type, image_url_dimensions)
-
-    return image_url
-
-
-def _get_preferred_image(image_type, dimension_list):
+    image_type = {
+        parse_qs(urlparse(img.get("$")).query)["t"][0]: {
+            "url": img.get("$"),
+            "description": img.get("description"),
```

Comment:
Using `dict::get` here makes me think it's possible that some of those values might not exist in the dictionary. But if not, then other aspects of this comprehension (e.g. `parse_qs(urlparse(None).query)["t"]`) will fail. We can probably simplify this to:
```suggestion
        parse_qs(urlparse(img["$"]).query)["t"][0]: {
            "url": img["$"],
            "description": img["description"],
```